hyperspy/_signals/electron_wave_image.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from hyperspy.signals import BaseSignal, Signal1D, ComplexSignal2D
+
+
+class ElectronWaveImage(ComplexSignal2D):
+
+    """ComplexSignal2D subclass for electron wave images."""
+
+    _signal_type = 'electron_wave'
+
+    def display_reconstruction_parameters(self):
```

Comment:
I am happy to merge this PR except for this method that seems to me half cooked as it is not very useful for n-dimensional signals. I imagine that it would be more useful a method that displays the parameters using the markers on a multi-dimensional figure. I suggest that we leave this class (`ElectronWaveImage`) for another pull request. Notice that, as the reconstructed waves will have "electron_wave" signal type, they'll be assigned this class once it is ready to merge in another PR.